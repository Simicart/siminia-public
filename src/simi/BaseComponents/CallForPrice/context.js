import React, { createContext, useContext } from 'react';
import { useCallForPrice } from './talons/useCallForPrice';

const CallForPriceContext = createContext();
const { Provider } = CallForPriceContext;

const CallForPriceProvider = props => {
    const { children } = props;

    const talonProps = useCallForPrice();
    const { error } = talonProps;

    if (error) {
        if (process.env.NODE_ENV !== 'production') {
            console.error(error);
        }
    }

    const contextValue = {
        ...talonProps
    };

    return <Provider value={contextValue}>{children}</Provider>;
};

export default CallForPriceProvider;

export const useCallForPriceContext = () => useContext(CallForPriceContext);
