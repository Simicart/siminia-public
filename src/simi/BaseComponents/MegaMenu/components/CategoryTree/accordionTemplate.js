import React from 'react';
import MenuTree from '../MenuTree';
import { useAccordionTemplate } from '../../talons/useAccordionTemplate';

const AccordionTemplate = () => {
  const { menuTree } = useAccordionTemplate();

  if (!menuTree) {
    return null;
  }

  return <MenuTree view="hamburger" menuTree={menuTree} />;
};

export default AccordionTemplate;
