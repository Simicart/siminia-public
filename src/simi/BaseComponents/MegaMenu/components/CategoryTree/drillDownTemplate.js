import React from 'react';
import { mergeClasses } from '@magento/venia-ui/lib/classify';
import { func, number, shape, string } from 'prop-types';
import { useDrillDownTemplate } from '../../talons/useDrillDownTemplate';
import { isEnableMenuItem } from '../../utils';
import defaultClasses from './drillDownTemplate.module.css';
import MenuLink from '../MenuTree/menuLink';

const DrillDownTemplate = props => {
  const { categoryId, onNavigate, setCategoryId, updateCategories } = props;

  const { categories, isShowIcons, isTopLevel } = useDrillDownTemplate({
    categoryId,
    updateCategories
  });
  const classes = mergeClasses(defaultClasses, props.classes);

  if (!categories) {
    return null;
  }

  const tree = categories.map(cat => {
    if (!isEnableMenuItem(cat.status, true)) {
      return null;
    }

    return (
      <li className={classes.item} key={cat.id}>
        <MenuLink
          view={'hamburger'}
          {...cat}
          onNavigate={onNavigate}
          toggleAction={setCategoryId}
          isShowIcons={isShowIcons}
          classes={{ toggle: classes.toggle, menuLink: classes.menuLink }}
          level={isTopLevel ? 1 : 2}
        />
      </li>
    );
  });

  return (
    <nav className={classes.root}>
      <ul className={classes.tree}>{tree}</ul>
    </nav>
  );
};

DrillDownTemplate.propTypes = {
  categoryId: number.isRequired,
  onNavigate: func.isRequired,
  setCategoryId: func.isRequired,
  updateCategories: func.isRequired,
  classes: shape({
    root: string
  })
};

export default DrillDownTemplate;
