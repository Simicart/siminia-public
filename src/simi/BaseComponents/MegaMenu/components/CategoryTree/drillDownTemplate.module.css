.item {
  composes: hamburger main from '../MenuTree/menuItem.module.css';
}

.toggle {
  composes: toggle from '../MenuTree/menuLink.module.css';
  transform: none;
}

.toggle svg {
  width: initial !important;
}

.menuLink {
  composes: menuLink from '../MenuTree/menuLink.module.css';
  height: 3.5rem;
}

.menuLink:not(.level1) {
  height: 3.5rem;
}
