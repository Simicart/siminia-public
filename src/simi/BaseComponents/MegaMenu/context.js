import React, { createContext, useContext } from 'react';
import { useMegaMenu } from './talons/useMegaMenu';

const AmMegaMenuContext = createContext();
const { Provider } = AmMegaMenuContext;

const AmMegaMenuProvider = props => {
    const { children } = props;

    const talonProps = useMegaMenu();
    const { error } = talonProps;

    if (error) {
        if (process.env.NODE_ENV !== 'production') {
            console.error(error);
        }
    }

    const contextValue = {
        ...talonProps
    };

    return <Provider value={contextValue}>{children}</Provider>;
};

export default AmMegaMenuProvider;

export const useAmMegaMenuContext = () => useContext(AmMegaMenuContext);
