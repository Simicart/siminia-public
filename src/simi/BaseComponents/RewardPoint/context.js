// import React, { createContext, useContext } from 'react';
// import { useRewardPoint } from './talons/useRewardPoint';

// const BssRewardPointContext = createContext();
// const { Provider } = BssRewardPointContext;

// const BssRewardPointProvider = props => {
//     const { children } = props;

//     const talonProps = useRewardPoint();
//     const { error } = talonProps;

//     if (error) {
//         if (process.env.NODE_ENV !== 'production') {
//             console.error(error);
//         }
//     }

//     const contextValue = {
//         ...talonProps
//     };

//     return <Provider value={contextValue}>{children}</Provider>;
// };

// export default BssRewardPointProvider;

// export const useAmMegaMenuContext = () => useContext(BssRewardPointContext);
