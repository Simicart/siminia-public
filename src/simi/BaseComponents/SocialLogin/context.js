import React, { createContext, useContext } from 'react';
import { useSocialLogin } from './talons/useSocialLogin';
import { useSignInEvent } from './talons/useSignInEvent';

const AmSocialLoginContext = createContext();
const { Provider } = AmSocialLoginContext;

const AmAdvancedReviewsProvider = props => {
  const { children } = props;

  const talonProps = useSocialLogin();
  const { error, needRedirect, redirectUrl, handleErrors } = talonProps;
  const { isBusy } = useSignInEvent({
    needRedirect,
    redirectUrl,
    handleErrors
  });

  if (error) {
    if (process.env.NODE_ENV !== 'production') {
      console.error(error);
    }

    console.log('Check if Amasty modules has been installed!');
  }

  const contextValue = {
    ...talonProps,
    isBusy
  };

  return <Provider value={contextValue}>{children}</Provider>;
};

export default AmAdvancedReviewsProvider;

export const useSocialLoginContext = () => useContext(AmSocialLoginContext);
