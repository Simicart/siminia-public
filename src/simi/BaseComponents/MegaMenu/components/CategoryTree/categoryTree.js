import React from 'react';
import { useAmMegaMenuContext } from '../../context';
import DefaultTemplate from '@magento/venia-ui/lib/components/CategoryTree/categoryTree';
import DrillDownTemplate from './drillDownTemplate';
import AccordionTemplate from './accordionTemplate';

const CategoryTree = props => {
  const { hamburgerView, isEnabledMegaMenu, isMobile } = useAmMegaMenuContext();

  if (!isEnabledMegaMenu) {
    return <DefaultTemplate {...props} />;
  } else if (isMobile && hamburgerView === 'drill') {
    return <DrillDownTemplate {...props} />;
  }

  return <AccordionTemplate {...props} />;
};

export default CategoryTree;
