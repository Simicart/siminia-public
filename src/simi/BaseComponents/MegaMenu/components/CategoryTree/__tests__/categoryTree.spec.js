import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import { useAmMegaMenuContext } from '../../../context';
import CategoryTree from '../categoryTree';

jest.mock(
  '@magento/venia-ui/lib/components/CategoryTree/categoryTree',
  () => 'DefaultTemplate'
);
jest.mock('../drillDownTemplate.js', () => 'DrillDownTemplate');
jest.mock('../accordionTemplate.js', () => 'AccordionTemplate');
jest.mock('../../../context', () => ({
  useAmMegaMenuContext: jest.fn().mockName('useAmMegaMenuContext')
}));

test('renders without data', () => {
  useAmMegaMenuContext.mockReturnValue({});

  const tree = createTestInstance(<CategoryTree />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('renders default template', () => {
  useAmMegaMenuContext.mockReturnValue({ isEnabledMegaMenu: false });

  const tree = createTestInstance(<CategoryTree />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('renders drill template', () => {
  useAmMegaMenuContext.mockReturnValue({
    isEnabledMegaMenu: true,
    hamburgerView: 'drill',
    isMobile: true
  });

  const tree = createTestInstance(<CategoryTree />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('renders accordion template', () => {
  useAmMegaMenuContext.mockReturnValue({
    isEnabledMegaMenu: true,
    hamburgerView: 'accordion'
  });

  const tree = createTestInstance(<CategoryTree />);
  expect(tree.toJSON()).toMatchSnapshot();
});
