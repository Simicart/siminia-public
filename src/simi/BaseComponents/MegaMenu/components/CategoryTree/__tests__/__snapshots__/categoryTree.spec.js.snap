// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders accordion template 1`] = `<AccordionTemplate />`;

exports[`renders default template 1`] = `<DefaultTemplate />`;

exports[`renders drill template 1`] = `<DrillDownTemplate />`;

exports[`renders without data 1`] = `<DefaultTemplate />`;
